addons/mail/static/src/core/web/message_patch.js

```diff
@@ -49,15 +49,15 @@ patch(Message.prototype, {
         };
     },
     hasAuthorClickable() {
-        return this.message.author?.main_user_id;
+        return this.message.author_id?.main_user_id;
     },
     onClickAuthor(ev) {
         if (this.hasAuthorClickable()) {
             markEventHandled(ev, "Message.ClickAuthor");
             const target = ev.currentTarget;
             if (!this.avatarCard.isOpen) {
                 this.avatarCard.open(target, {
-                    id: this.message.author.main_user_id.id,
+                    id: this.message.author_id?.main_user_id?.id,
```

Comment:
adding `?` is not necessary here, it makes no sense to call this function without an id
check is already done inside `hasAuthorClickable`